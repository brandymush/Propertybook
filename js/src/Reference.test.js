// @flow
import React from 'react';
import warning from 'warning';
import { render } from '@testing-library/react';

// Public API
import { Reference } from '.';

// Private API
import { ManagerReferenceNodeSetterContext } from './Manager';

jest.mock('warning');

describe('Arrow component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the expected markup', () => {
    const setReferenceNode = jest.fn();

    const { asFragment } = render(
      <ManagerReferenceNodeSetterContext.Provider value={setReferenceNode}>
        <Reference>{({ ref }) => <div ref={ref} />}</Reference>
      </ManagerReferenceNodeSetterContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('consumes the ManagerReferenceNodeSetterContext from Manager', () => {
    const setReferenceNode = jest.fn();

    render(
      <ManagerReferenceNodeSetterContext.Provider value={setReferenceNode}>
        <Reference>{({ ref }) => <div ref={ref} />}</Reference>
      </ManagerReferenceNodeSetterContext.Provider>
    );
    expect(setReferenceNode).toHaveBeenCalled();
  });

  it('warns when setReferenceNode is present', () => {
    const setReferenceNode = jest.fn();

    render(
      <ManagerReferenceNodeSetterContext.Provider value={setReferenceNode}>
        <Reference>{({ ref }) => <div ref={ref} />}</Reference>
      </ManagerReferenceNodeSetterContext.Provider>
    );
    expect(warning).toHaveBeenCalledWith(
      true,
      '`Reference` should not be used outside of a `Manager` component.'
    );
  });

  it('does not warn when setReferenceNode is not present', () => {
    render(
      <ManagerReferenceNodeSetterContext.Provider value={undefined}>
        <Reference>{({ ref }) => <div ref={ref} />}</Reference>
      </ManagerReferenceNodeSetterContext.Provider>
    );
    expect(warning).toHaveBeenCalledWith(
      false,
      '`Reference` should not be used outside of a `Manager` component.'
    );
  });
});
