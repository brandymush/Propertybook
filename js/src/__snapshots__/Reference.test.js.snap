// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component renders the expected markup 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
